mammoth/bin/translate.py

```diff
@@ -26,12 +28,24 @@ def translate(opts):
     if 'adapters' in corpus_opts:
         encoder_adapter_ids = get_adapter_ids(opts, corpus_opts, 'encoder')
         decoder_adapter_ids = get_adapter_ids(opts, corpus_opts, 'decoder')
+        uses_adapters = True
     else:
         encoder_adapter_ids = None
         decoder_adapter_ids = None
+        uses_adapters = False
+
+    node_rank = 0
+    local_rank = 0
+    if use_gpu(opts):
+        context_enum = DeviceContextEnum.SINGLE_GPU
```

Comment:
so no sharding supported in decoding? lgtm

Reply:
Automatic sharding would be nice for usability, but hasn't been implemented.
As a workaround, you can shard in preprocessing, translate each shard in parallel, and then rejoin.